test/plugin_functional/plugins/doc_views_links_plugin/public/plugin.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import angular from 'angular';
+import React from 'react';
+import { Plugin, CoreSetup } from 'opensearch-dashboards/public';
+import { DiscoverSetup } from '../../../../../src/plugins/discover/public';
+
+angular.module('myDocViewLink', []).directive('myHit', () => ({
+  restrict: 'E',
+  scope: {
+    hit: '=hit',
+  },
+  template: '<h1 data-test-subj="href-docviewlink">{{hit._index}}</h1>',
+}));
+
+function MyHit(props: { index: string }) {
+  return <h1 data-test-subj="generateUrlFn-docviewlink">{props.index}</h1>;
+}
+
+export class DocViewsLinksPlugin implements Plugin<void, void> {
+  public setup(core: CoreSetup, { discover }: { discover: DiscoverSetup }) {
+    discover.docViewsLinks.addDocViewLink({
+      href: 'http://locationHref',
+      order: 1,
+      label: 'href doc view link',
+    });
+
+    discover.docViewsLinks.addDocViewLink({
+      generateUrlFn: (props) => 'http://locationHref',
```

Comment:
This might be caught in the link checker failure. Could use any one of this urls: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.lycheeexclude, or add this.

Reply:
Thanks, used a different link. Also, added proper e2e to this flow.